mdstat.go

```diff
@@ -112,39 +143,52 @@ func parseMDStat(mdstatData []byte) ([]MDStat, error) {
 	return mdStats, nil
 }
 
-func evalStatusLine(statusline string) (active, total, size int64, err error) {
-	matches := statuslineRE.FindStringSubmatch(statusline)
-	if len(matches) != 4 {
-		return 0, 0, 0, fmt.Errorf("unexpected statusline: %s", statusline)
-	}
+func evalStatusLine(lines []string, i int) (active, total, size int64, err error) {
```

Comment:
I think this would be better if it took two string parameters like `deviceLine` and `statusLine` to make it more readable and avoid any potential index out of bounds.

Reply:
Good point